### ISSUE TITLE
V3 sensor.vehicle_energy_added incompatible with energy dashboard

### ISSUE THREAD

**Issue by purcell-lab** [alandtse/tesla#295]
## Version of the custom_component

3.0.0

## Describe the bug

I understand the vehicle _energy_added sensor is a bit of a cludge. I'm raising this issue to highlight it's incompatibility with the HA energy dashboard.

Here is my sensor for today:

![Screenshot_20221023-111728](https://user-images.githubusercontent.com/79175134/197368547-e4f57dba-1d5d-4199-bc1d-68260027177a.png)

Looks like I have consumed about 35 kWh for the day.

Energy dashboard is however reporting 135 kWh for the day.

![Screenshot_20221023-111841](https://user-images.githubusercontent.com/79175134/197368579-24d72d64-57b1-4961-9c6e-767d64d7ace4.png)

It is doing this as the energy sensor is resetting to 0 between charges and the energy dashboard double counts every time it rests to zero.

I suspect it maybe required to make a smarter energy sensor to support this integration and the HA energy dashboard.

**Comment by alandtse** [comment]
HA currently does not expect something that does energy to be able to move. So yes, we probably need a specific sensor anyway so any energy_added only is counted when at a home location or HA needs to add support for it directly. PRs are welcome.

**Comment by craigsbits** [comment]
I am sure this was working in previous version.  It had a slightly different name though with sensor at end - sensor."vehicle name"_energy_added_sensor << .  The "sensor" reset when car was unplugged but was additive for all discrete charges during the night.

**Comment by TheGreatBorg** [comment]
It is nice to see and follow the "charging flow" but the figure is only showing when the car is in Charging mode (but 90% of the time it is zero). 

We (I) really miss the "last added energy" that we could add to a multi-meter helper for what was charged at the last charging session.

**Comment by craigsbits** [comment]
Interesting that the added range seems ok so it is the energy sensor thats at fault.

**Comment by TheGreatBorg** [comment]
Well I see that we might have a conflict of interest :-) We try to solve two things with one sensor.

1. Show/follow charging flow when charging
2. Keep track of total added energy after each charging cycle

I don't think that multi-meter option can manage the increasing sensor value (it can't if you take a temperature sensor and put into a multi-meter helper - delta option can't even help). 

For now it is quite hard to figure out the max value added from last charging cycle for us to keep and save for the measuring daily added energy (compared to daily driven ml/km). The sensor goes to zero when charging has stopped and if we are not connected to the car we will fetch the event (OnBeforeChargingStops).

I hope my confusing explanation is OK (English is not my first language)

The v.2 Energy Added was way more useful than this "Current adding Energy" value.

**Comment by TheGreatBorg** [comment]
I would very much like to help but sadly I can't. I cannot build the integration_blueprint Dev-container on my Windows 11 PC. 

I have forked and cloned the project and from what I can see the solution would be adding a property to the TeslaCarChargerEnergy Class. It should have an extra attribute "added_energy" that can be compared with the "added_range" attribute. This "added_energy" value will be set to Zero (reset) when charging_state == "Charging" and when charging has ended the added_energy should be equal to Energy_added. Then we would have a multi purpose  charging sensor.

**Comment by metrafonic** [comment]
@TheGreatBorg One thing i slightly likes with earlier versions of the addon were that the "energy_added" would reset to 0 kwh only when the charging cable went from disconnected to connected. A charging session was the whole time a charger was connected, (but the value of the charging session was still available after disconnecting the cable).

I am for example using a https://www.home-assistant.io/integrations/utility_meter/ to measure how much my vehicle charges at different locations, so i for example dont exceed my home charging quota. 

With the current state of the energy_added sensor, the reset to 0 will happen **before** the last value has been updated (sensor is updated every 5 minutes or so), so the peak measurement is always around 1-4kwh short of the real value. Now i have to manually force update before stopping to charge. The sensor should retrieve the last known charge value from the tesla api, save this value to the sensor state __and then__ set it to 0.

TLDR, there needs to be done more work
